src/signalduino.h

```diff
@@ -97,29 +97,29 @@ void setup() {
   }
 
   // defined states - pullup on for unused pins
-  #ifndef ARDUINO_RADINOCC1101 // not RADINOCC1101
-    #ifdef CMP_CC1101
-      uint8_t offset = 0; // different start of the pins
-      #ifdef ARDUINO_ATMEGA328P_MINICUL
-        offset = 1;
-      #endif
-      
-      for (uint8_t i=4 + offset;i<9 + offset;i++) {
-        pinAsInputPullUp(i);
-      }
-      for (uint8_t i=14 + offset;i<20;i++) {
-        pinAsInputPullUp(i);
-      }
-    #else // Arduino Nano without CC1101
-      for (uint8_t i=3;i<20;i++) { // not PIN_RECEIVE
-        if ( i != PIN_SEND && i != PIN_LED ) {pinAsInputPullUp(i);}
-      }
+  // Start behind RX / TX Pin´s --> all hardware used for system serial 0 & 1
+  // End on Pin 23 --> radino with used highest Pin numbre
+  for (uint8_t i=2;i<24;i++) {
+    // pins for receive, send, LED
+    if (  (i == PIN_RECEIVE) || (i == PIN_SEND) || (i == PIN_LED) ) { continue; }
+
+    // cc1101 pins
+    #if defined(CMP_CC1101)
+      if ((i == csPin) || (i == mosiPin) || (i == misoPin) || (i == sckPin)) { continue; }
     #endif
-  #else // everything except RADINOCC1101
-    for (uint8_t i=2;i<24;i++) {
-      if (i != 4 && ( (i < 7 || i > 9 ) && (i < 13 || i > 17) ) ) {pinAsInputPullUp(i);} // not PIN_Mark,GDO2,SS,GDO0,LED
-    }
-  #endif
+
+    // PIN_MARK433 pin (only on RADINOCC1101 & ATMEGA328P_MINICUL)
+    #if defined(ARDUINO_RADINOCC1101) || defined(ARDUINO_ATMEGA328P_MINICUL)
+      if (i == PIN_MARK433) { continue; }
+    #endif
+
+    // more pins´s (only on RADINOCC1101)
+    #if not defined (ARDUINO_RADINOCC1101)
+      if (i >= 20) { continue; }
```

Comment:
Hier würde ich break verwenden, sobald wir keine Pins mehr initialisieren wollen.
Sieht für mich insgesamt jetzt stimmig aus.

Leider wird es uns ein paar Byte an Flash kosten :(

Reply:
Wir werden uns nie in deinen Kopf denken :D Code folgt morgen wenn dir break besser gefällt.

Flash, da haben wir uns nicht einen Überblick verschafft. Die Funktion war hier wichtiger.

Reply:
Break bricht die Schleife ab.
Continue führt ja nur zum Nächsten Durchlauf.

Die Funktion ist hier wichtiger ja.

Reply:
Waere es im Grunde nicht besser und einfacher explizit *alle* Pins erstmal als Input mit Pullups zu setzen und dann spaeter, je nach Controller, die richtigen Pins auf Ausgang bzw. Eingang ohne Pullup (also MISO als Eingang, die anderen SPI Ports als Ausgang, usw.)

Letztlich sind ja gerade die Pins, die verwendet werden, bekannt ueber die jeweiligen Defines.
Dann kann man ueber die entsprechenden Namen einfach die Pins richtig setzen. Der Code sollte dann hier auch nur einmal exisitieren (wenn die Defines fuer alle Controller gleich heissen, wenn nicht, sollte man hier ansetzen).

Auch wenn klar ist, was hier in den Schleifen passiert - es ist irgendwie unleserlich und in 2 Jahren versteht das niemand mehr, ohne sich tief reinzudenken.

Daher lieber sowas:
for(uint8_t i=0; i <= MAXIONUM /* muss man je nach plattform natuerlich haben, koennte aber auch ein define werden */; ++i)
{
  pinAsInputPullUp(i); 
}

#ifdef CC1101
pinAsOutput(MOSI);
pinAsInput(MISO);
#else
pinAsOutput(IRGENDEINANDERERPIN)
pinAsInput(UNDDERHIERALSEINGANG)
#endif
...

(Habe gerade nicht nachgeschaut, wie die Funktionen genau heissen oder welche Pins es genau gibt - geht ums Prinzip)
Sollte - wenn die defines passen - dann fuer alle Controller passen. Weiter verfeinern ueber #ifdefs geht natuerlich.

Generell: Code wird oefters gelesen, als geschrieben.

Reply:
> Waere es im Grunde nicht besser und einfacher explizit _alle_ Pins erstmal als Input mit Pullups zu setzen und dann spaeter, je nach Controller, die richtigen Pins auf Ausgang bzw. Eingang ohne Pullup (also MISO als Eingang, die anderen SPI Ports als Ausgang, usw.)

So war es ja ursprünglich im Prinzip, aber das führte zu Fehlfunktionen.